From 71e764952dcb39a053294675565ea7942dd6a923 Mon Sep 17 00:00:00 2001
From: Peter Bergin <peter.bergin@windriver.com>
Date: Thu, 25 May 2023 15:05:14 +0200
Subject: [PATCH 1/2] Cargo.toml: remove panic setting

rustlibs in is built with panic strategy 'unwind'.
meta/classes-recipe/rust-common.bbclass and the variable
RUST_PANIC_STRATEGY. Don't know exactly how to handle this specific
case with FFI but to be able to build remove explicit panic strategy
and rely on default.

Upstream-Status: Pending

Signed-off-by: Peter Bergin <peter.bergin@windriver.com>
---
 Cargo.toml | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index afcdc993..c65bc842 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -111,9 +111,6 @@ path = "rust/src/main.rs"
 opt-level = 1 # No optimizations are too slow for us.
 
 [profile.release]
-# Unwinding across FFI boundaries is undefined behavior, and anyways, we're
-# [crash-only](https://en.wikipedia.org/wiki/Crash-only_software)
-panic = "abort"
 # We assume we're being delivered via e.g. RPM which supports split debuginfo
 debug = true
 
-- 
2.34.1

